src/mon/Monitor.cc

```diff
@@ -1776,7 +1776,8 @@ void Monitor::handle_probe_reply(MonOpRequestRef op)
 	   << " upgrading" << dendl;
       exit(0);
     }
-    if (!osdmon()->osdmap.test_flag(CEPH_OSDMAP_PURGED_SNAPDIRS)) {
+    if (!osdmon()->osdmap.test_flag(CEPH_OSDMAP_PURGED_SNAPDIRS) ||
+	!osdmon()->osdmap.test_flag(CEPH_OSDMAP_RECOVERY_DELETES)) {
```

Comment:
Is RECOVERY_DELETES actually set as part of a scrub? I don't know how that works.

Reply:
807c4c834136e5a3d20b31d50e770430695473ff